### ISSUE TITLE
No secondary market for position tokens

### ISSUE THREAD

**Issue by andrei-anisimov** [carboclan/dapp.honeylemon.market#111]
In the current implementation the orderbook holds orders that represent an offer to enter a contract that is active at the time the order is filled by the taker (current day contract). Current day MarketProtocol ERC20 position tokens are minted at the time of the order fill. There's currently no secondary market/orderbook for those position tokens - users and market makers alike have to hold those tokens until expiration in order to settle them. 

This creates a capital efficiency and balance sheet risk challenges for market makers, since there's no way for them to offload tokens from the balance sheet. The only way a market maker can exit their position early is if during the same day they entered the position they enter an opposite position and trigger an early redemption in the market protocol by supplying both short and long position tokens - we need to think through the viability of this mechanism alone and whether another mechanism/market/orderbook is required.

**Comment by jie85** [comment]
I think the redemption method should works. Thus, 1. the order book is still for minting tokens only 2. the market maker will get new tokens from the order book when his orders are filled. 3. The market maker redeems the tokens when he has long and short tokens simultaneously, which increases his capital efficiency.

**Comment by andrei-anisimov** [comment]
So in your view this mechanism alone is sufficient (no need for a secondary order book?)

**Comment by jie85** [comment]
Yes. I think it is sufficient for the market maker. And the secondary order book mechanism is a little too complicated.
